src/Chainweb/Payload/PayloadStore/InMemory.hs

```diff
@@ -57,8 +57,7 @@ newPayloadCache = PayloadCache <$> newBlockOutputsStore
     <*> newTransactionTreeStore
     <*> newOutputTreeStore
 
-newPayloadDb :: IO (PayloadDb HashCAS.HashMapCas)
-newPayloadDb = PayloadDb
-    <$> newTransactionDb
-    <*> newPayloadCache
-
+newPayloadDb :: IO (Maybe (PayloadDb HashCAS.HashMapCas))
+newPayloadDb = do
+    pdb <- PayloadDb <$> newTransactionDb <*> newPayloadCache
+    return $ Just pdb
```

Comment:
Why does this need to return in `Maybe`?

Reply:
It was done as a Maybe because none of the mempool tests (except for the one I added) needed it, and it looked like a lot of work to contruct dummy instance for all the other tests.  I'll look at it again, I maybe have an easier time after the past few days experiences.